src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -106,13 +124,20 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
             {
                 if (mapIndices.DataType != typeof(long))
                     throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
-                if (mapIndices.Length != Length)
-                    throw new ArgumentException(Strings.MismatchedColumnLengths, nameof(mapIndices));
+                if (mapIndices.Length > Length)
+                    throw new ArgumentException(Strings.MapIndicesExceedsColumnLenth, nameof(mapIndices));
                 return Clone(mapIndices as PrimitiveColumn<long>, invertMapIndices);
             }
             return Clone();
         }
 
+        public override BaseColumn CloneAndAppendNulls(long numberOfNulls, BaseColumn mapIndices = null, bool invertMapIndices = false)
+        {
+            PrimitiveColumn<T> ret = Clone(mapIndices, invertMapIndices) as PrimitiveColumn<T>;
```

Comment:
Add `Debug.Assert(ret != null);` just to be on the safe side?

And rename `ret` to `clone`?